package/android/src/main/java/com/shopify/reactnative/skia/SkiaBaseView.java

```diff
@@ -175,9 +144,9 @@ public void onSurfaceTextureSizeChanged(SurfaceTexture surface, int width, int h
     public boolean onSurfaceTextureDestroyed(SurfaceTexture surface) {
         Log.i("SkiaBaseView", "onSurfaceTextureDestroyed");
         // https://developer.android.com/reference/android/view/TextureView.SurfaceTextureListener#onSurfaceTextureDestroyed(android.graphics.SurfaceTexture)
-        surfaceDestroyed();
-        mSurface.release();
-        mSurface = null;
+        destroySurface();
+        // We create again a new SurfaceTexture to have a faster time to first frame (if API >= 26)
+        createSurfaceTexture();
```

Comment:
IMO it'd be more intuitive to manage surface lifecycle from [onAttachToWIndow](https://developer.android.com/reference/android/view/View#onAttachedToWindow()) and [onDetachedFromWindow ](https://developer.android.com/reference/android/view/View#onDetachedFromWindow()). 

Otherwise this looks fine. One thing in this scenario is that it seem like you'd create a new surface texture for the view that gets unmounted. Since it is done in destroy logic you can't tell if the view will be reused later or just completely removed from hierarchy.